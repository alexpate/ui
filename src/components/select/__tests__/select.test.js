/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import Select from '../select';

expect.addSnapshotSerializer(serializer);

describe('components/select', () => {
  describe('Select (default)', () => {
    const create = (spread = {}) => {
      const props = {
        // (insert your default props here)
        ...spread,
      };
      const element = <Select {...props} />;
      return {props, element};
    };

    it('should shallow render as expected', () => {
      const {element} = create({
        // (insert test specific props here)
        children: 'Child',
      });
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });
  });
});
