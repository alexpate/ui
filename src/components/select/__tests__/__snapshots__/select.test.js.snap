// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render a disabled select 1`] = `
<div
  className="css-tvxmfb0 css-1s58okm"
  disabled={true}
  onSelect={[Function]}
  open={false}
  style={Object {}}
>
  <div
    className="css-hvar41 css-tvxmfb5"
    onClick={undefined}
  >
    <span
      className="css-16ef8bd0 css-1rchwux"
      color="navy500"
      onClick={undefined}
      size="small"
      style={Object {}}
    >
      Select…
    </span>
    <div
      className="css-elkfk7 css-tvxmfb1"
      open={false}
    >
      <i
        className="css-fj83eq0 css-13jt8c1"
        color="grey500"
        onClick={undefined}
        size={24}
        style={Object {}}
        title={undefined}
      >
        <svg
          aria-hidden="true"
          height={24}
          width={24}
        >
          <title>
            chevron_right
          </title>
          <use
            xlinkHref="#chevron_right"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="css-17y3pak css-tvxmfb3"
    onClick={[Function]}
    open={false}
  >
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hertfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Bedfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Oxfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hampshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;

exports[`Select should render a readonly select 1`] = `
<div
  className="css-tvxmfb0 css-umjnfj"
  disabled={false}
  onSelect={[Function]}
  open={false}
  style={Object {}}
>
  <div
    className="css-hvar41 css-tvxmfb5"
    onClick={undefined}
  >
    <span
      className="css-16ef8bd0 css-1rchwux"
      color="navy500"
      onClick={undefined}
      size="small"
      style={Object {}}
    >
      Select…
    </span>
    <div
      className="css-elkfk7 css-tvxmfb1"
      open={false}
    >
      <i
        className="css-fj83eq0 css-13jt8c1"
        color="grey500"
        onClick={undefined}
        size={24}
        style={Object {}}
        title={undefined}
      >
        <svg
          aria-hidden="true"
          height={24}
          width={24}
        >
          <title>
            chevron_right
          </title>
          <use
            xlinkHref="#chevron_right"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="css-17y3pak css-tvxmfb3"
    onClick={[Function]}
    open={false}
  >
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hertfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Bedfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Oxfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hampshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;

exports[`Select should render shallow component ok 1`] = `
<div
  className="css-tvxmfb0 css-1jyqghw"
  disabled={false}
  onSelect={[Function]}
  open={false}
  style={Object {}}
>
  <div
    className="css-hvar41 css-tvxmfb5"
    onClick={[Function]}
  >
    <span
      className="css-16ef8bd0 css-1bmxkxu"
      color="navy700"
      onClick={undefined}
      size="small"
      style={Object {}}
    >
      Select…
    </span>
    <div
      className="css-elkfk7 css-tvxmfb1"
      open={false}
    >
      <i
        className="css-fj83eq0 css-1hy8mwc"
        color="navy700"
        onClick={undefined}
        size={24}
        style={Object {}}
        title={undefined}
      >
        <svg
          aria-hidden="true"
          height={24}
          width={24}
        >
          <title>
            chevron_right
          </title>
          <use
            xlinkHref="#chevron_right"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="css-17y3pak css-tvxmfb3"
    onClick={[Function]}
    open={false}
  >
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hertfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hertfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Bedfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Bedfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Oxfordshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Oxfordshire
      </div>
    </div>
    <div
      className="css-14jjchb css-tvxmfb4"
      name="Hampshire"
      onClick={[Function]}
      selected={undefined}
    >
      <div>
        Hampshire
      </div>
    </div>
  </div>
</div>
`;
